### PR TITLE
12.0 mig sale disable inventory check

### PR DESCRIPTION
**Pull request by Montxu**
cc @guadaltech

### CODE REVIEW

**Review by rousseldenis — APPROVED** [review]
Code review.

I know this is a migration, but usually it's better to have an option to enable this as it breaks the standard flow (it's wanted I know) and could break other tests. Maybe put in ROADMAP.

File: sale_disable_inventory_check/tests/test_disable_inventory_check.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+class TestDisableInventoryCheck(common.TransactionCase):
+
+    def setUp(self):
+        super(TestDisableInventoryCheck, self).setUp()
+
+        # Create stockable product
+        self.product_1 = self.env['product.product'].create({
+            'name': 'Product 1',
+            'type': 'product',
+            'uom_id': self.env.ref('product.product_uom_unit').id,
```
Comment (nikul-serpentcs) [inline sale_disable_inventory_check/tests/test_disable_inventory_check.py]: @Montxu `self.env.ref('product.product_uom_unit')` --> `self.ref('uom.product_uom_unit')`

**Review by nikul-serpentcs — CHANGES_REQUESTED** [review]
Improved Code

**Review by HaraldPanten — APPROVED** [review]
Functional review 👍

**Comment by pedrobaeza** [comment]
> I know this is a migration, but usually it's better to have an option to enable this as it breaks the standard flow (it's wanted I know) and could break other tests. Maybe put in ROADMAP.

The way to do it is via context and config check:
```
from odoo.tools import config

    @api.onchange('product_uom_qty', 'product_uom', 'route_id')
    def _onchange_product_id_check_availability(self):
        if (config['test_enable'] and 
                not self.env.context.get('test_sale_disable_inventory_check')):
            return super()._onchange_product_id_check_availability()
        return {}
```

and put that context on current test.

**Comment by Montxu** [comment]
Fixed, @nikul-serpentcs. Are the changes OK?

**Review by nikul-serpentcs — APPROVED** [review]
only Code review LGTM :+1:

**Comment by rousseldenis** [comment]
> The way to do it is via context and config check:

I don't like this way. You cannot put code that depends on tests...

**Comment by pedrobaeza** [comment]
Denis, that's done several times in Odoo and in OCA. It's not ideal but it allows you a good way to control this kind of incompatibilities

**Comment by rousseldenis** [comment]
> that's done several times in Odoo and in OCA

Why not another way of thinking ? :smiley: 

> It's not ideal but it allows you a good way to control this kind of incompatibilities

That's why an option is more convenient.

I'm not blocking on this but remind it's not a good way of coding.

**Comment by pedrobaeza** [comment]
Your alternative seems to me more convoluted and requires more configuration

**Review by legalsylvain — APPROVED** [review]
thanks for porting this module ! 
LGTM. code review / not test.

**Pull request merged**